embeddings/task/lightning_task/text_classification.py

```diff
@@ -0,0 +1,98 @@
+from torch.utils.data import DataLoader
+from torchmetrics import F1, Accuracy, MetricCollection, Precision, Recall
+from transformers import AutoModelForSequenceClassification
+
+from embeddings.data.datamodule import HuggingFaceDataset
+from embeddings.task.lightning_task.lightning_task import HuggingFaceLightningTask
+
+
+class TextClassification(HuggingFaceLightningTask):
+    downstream_model_type = AutoModelForSequenceClassification
+
+    def __init__(
+        self,
+        num_labels: int,
+        model_name_or_path: str,
+        unfreeze_transformer_from_layer: Optional[int] = None,
+        metrics: Optional[MetricCollection] = None,
+        config_kwargs: Optional[Dict[str, Any]] = None,
+        task_model_kwargs: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        super().__init__(
+            num_labels=num_labels,
+            model_name_or_path=model_name_or_path,
+            downstream_model_type=self.downstream_model_type,
+            unfreeze_transformer_from_layer=unfreeze_transformer_from_layer,
+            metrics=metrics,
+            config_kwargs=config_kwargs,
+            task_model_kwargs=task_model_kwargs,
+        )
+
+    def get_default_metrics(self) -> MetricCollection:
+        if self.hparams.num_labels > 2:
+            metrics = MetricCollection(
+                [
+                    Accuracy(num_classes=self.hparams.num_labels),
+                    Precision(num_classes=self.hparams.num_labels, average="macro"),
+                    Recall(num_classes=self.hparams.num_labels, average="macro"),
+                    F1(num_classes=self.hparams.num_labels, average="macro"),
+                ]
+            )
+        else:
+            metrics = MetricCollection(
+                [
+                    Accuracy(num_classes=self.hparams.num_labels),
+                    Precision(num_classes=self.hparams.num_labels),
+                    Recall(num_classes=self.hparams.num_labels),
+                    F1(num_classes=self.hparams.num_labels),
+                ]
+            )
+        return metrics
+
+    def forward(self, *args: Any, **kwargs: Any) -> Any:
+        return self.model(*args, **kwargs)
+
+    def shared_step(self, **batch: Any) -> Tuple[torch.Tensor, torch.Tensor]:
+        outputs = self.forward(**batch)
+        loss, logits = outputs[:2]
+        return loss, logits
+
+    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:
+        batch, batch_idx = args
+        loss, preds = self.shared_step(**batch)
+        self.train_metrics(preds, batch["labels"])
+        self.log("train/Loss", loss, on_step=True, on_epoch=True)
+        if self.hparams.use_scheduler:
+            last_lr = self.lr_scheduler["scheduler"].get_last_lr()
+            self.log("train/BaseLR", last_lr[0], on_step=True, on_epoch=True, prog_bar=True)
+            self.log("train/LambdaLR", last_lr[1], on_step=True, on_epoch=True, prog_bar=True)
```

Comment:
It should be `on_epoch=True`because it is simply on a step, not epoch :D however, according to https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#log, it should be set by default but I have never tried.